File: Problem_set_1/Introductory_problems/p07_Two_Knights/TwoKnights.java
import java.util.Scanner;

public class TwoKnights {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
    
        long n = scan.nextLong();

        for(long j = 1; j <= n; j++){
            long totalWays = j * j * (j * j - 1) / 2;
            long attackingWays = 4 * (j - 1) * (j - 2);

            long result = totalWays - attackingWays;
            System.out.println(result);
        }
        scan.close();
    }
}

File: Problem_set_1/Introductory_problems/p07_Two_Knights/TwoKnights.cpp
#include<iostream>

using namespace std;

int main(){
    long long n;
    cin >> n;

    for(int i = 1; i <= n; i++){
        long long totalWays = i * i * (i * i - 1) / 2;
        long long attackingWays = 4 * (i - 1) * (i - 2);

        long long result = totalWays - attackingWays;
        cout << result;
    }

    return 0;
}